xds/internal/balancer/cdsbalancer/cdsbalancer.go

```diff
@@ -206,16 +206,8 @@ func (b *cdsBalancer) handleSecurityConfig(config *xdsresource.SecurityConfig) e
 
 	}
 
-	bc := b.xdsClient.BootstrapConfig()
-	if bc == nil || bc.CertProviderConfigs == nil {
```

Comment:
We got rid of the nil check for the whole bootstrap config alongside the CertProviderConfig check (the point of this PR). We also pass a nil bootstrap to buildProvider. Is this intended or do we want to leave this nil check?

Reply:
We fail the bootstrap process if we do not have config that we process successfully. This `bc == nil` is probably a vestigial check. We don't need it anymore. IIRC, sometime in the past, we did have tests where the bootstrap config was `nil` when we used the fake xds client. Currently, there are very few tests, if at all any, that use the fake xds client, and we have methods in the `xdsclient` package to create an xds client with config for tests.